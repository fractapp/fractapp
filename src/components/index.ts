export {BlueButton} from './BlueButton';
export {WhiteButton, Img} from './WhiteButton';
export {SeedButton} from './SeedButton';
export {StatisticsBar} from './StatisticsBar';
export {PasswordInput} from './PasswordInput';
export {TextInput} from './TextInput';
export {TransactionInfo} from './TransactionInfo';
export {WalletInfo} from './WalletInfo';
export {WalletDetailsInfo} from './WalletDetailsInfo';
export {Dialog} from './Dialog';
export {Loader} from './Loader';
export {DriveItemView} from './DriveItemView';
export {PassCode} from './PassCode';
export {SuccessButton} from './SuccessButton';
export {WalletLogo} from './WalletLogo';
export {TabMenu} from './TabMenu';
export {ChatShortInfo} from './ChatShortInfo';
export {PaymentMsg} from './PaymentMsg';
export {Contact} from './Contact';
export {Receiver} from './Receiver';
export {AmountInput} from './AmountInput';
export {AmountValue} from './AmountValue';
export {ReceiverWithEnterAddress} from './ReceiverWithEnterAddress';
