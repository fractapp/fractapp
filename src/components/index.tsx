export {BlueButton} from './BlueButton';
export {WhiteButton, Img} from './WhiteButton';
export {SeedButton} from './SeedButton';
export {StatisticsBar} from './StatisticsBar';
export {PasswordInput} from './PasswordInput';
export {TextInput} from './TextInput';
export {TransactionInfo} from './TransactionInfo';
export {WalletInfo} from './WalletInfo';
export {WalletDetailsInfo} from './WalletDetailsInfo';
export {Dialog} from './Dialog';
export {Loader} from './Loader';
export {DriveItemView} from './DriveItemView';
export {PassCode} from './PassCode';
export {SuccessButton} from './SuccessButton';
export {WalletLogo} from './WalletLogo';
