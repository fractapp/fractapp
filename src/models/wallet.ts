import { ImageSourcePropType } from "react-native";

export enum Currency {
    Polkadot = 0,
    Kusama
}

export function getLogo(currency: Currency) {
    let logo: ImageSourcePropType
    switch (currency) {
        case Currency.Polkadot:
            logo = require("assets/img/dot.png")
            break
        case Currency.Kusama:
            logo = require("assets/img/kusama.png")
            break
        default:
            throw ("invalid currency")
    }
    return logo
}
export function getSymbol(currency: Currency) {
    let symbol = ""
    switch (currency) {
        case Currency.Polkadot:
            symbol = "DOT"
            break
        case Currency.Kusama:
            symbol = "KSM"
            break
        default:
            throw ("invalid currency")
    }
    return symbol
}

export class Wallet {
    name: string;
    address: string;
    balance: number;
    currency: Currency;
    usdValue: number;

    constructor(name: string, address: string, currency: Currency, balance: number, price: number) {
        this.name = name;
        this.address = address;
        this.balance = balance;
        this.currency = currency;

        this.usdValue = +(balance * price).toFixed(2)
    }
}
