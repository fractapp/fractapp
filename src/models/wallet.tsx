import { ImageSourcePropType } from "react-native";

export enum Currency {
    Polkadot = 0,
    Kusama
}

export class Wallet {
    name: string;
    address: string;
    balance: number;
    currency: Currency;

    symbol: string;
    usdValue: number;
    logo: ImageSourcePropType;

    constructor(name: string, address: string, currency: Currency, balance: number) {
        this.name = name;
        this.address = address;
        this.balance = balance;
        this.currency = currency;
        
        //TODO get state
        let rate = 10
        this.usdValue = balance * rate
        switch (currency) {
            case Currency.Polkadot:
                this.logo = require("assets/img/dot.png")
                this.symbol = "DOT"
                break
            case Currency.Kusama:
                this.logo = require("assets/img/kusama.png")
                this.symbol = "KSM"
                break
        }

    }
}
