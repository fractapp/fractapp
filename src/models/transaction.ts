import {Currency} from './wallet';

/**
 * Transaction type
 * @category Models
 */
export enum TxType {
  None = 0,
  Sent,
  Received,
}

export enum TxStatus {
  Pending = 0,
  Success,
  Fail,
}

/**
 * @category Models
 */
export class Transaction {
  id: string;
  member: string;
  currency: Currency;
  value: number;
  txType: TxType;
  timestamp: number;
  fee: number;
  usdValue: number;
  usdFee: number;
  status: TxStatus;

  constructor(
    id: string,
    member: string,
    currency: Currency,
    txType: TxType,
    timestamp: number,
    value: number,
    usdValue: number,
    fee: number,
    usdFee: number,
    status: TxStatus,
  ) {
    this.id = id;
    this.member = member;
    this.currency = currency;
    this.value = value;
    this.txType = txType;
    this.timestamp = timestamp;
    this.fee = fee;
    this.usdValue = usdValue;
    this.usdFee = usdFee;
    this.status = status;
  }
}
