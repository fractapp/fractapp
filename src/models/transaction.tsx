import { ImageSourcePropType } from "react-native";
import { Currency } from "./wallet";

export enum TxType {
    None = 0,
    Sent,
    Received
}

export class Transaction {
    id: string;
    member: string;
    currency: Currency;
    value: number;
    txType: TxType;
    date: Date;

    usdValue: number;
    symbol: string;
    logo: ImageSourcePropType;

    constructor(id: string, member: string, currency: Currency, txType: TxType, date: Date, value: number) {
        this.id = id;
        this.member = member;
        this.currency = currency;
        this.value = value;
        this.txType = txType;
        this.date = date;
        //TODO get state
        let rate = 10
        this.usdValue = value * rate
        switch (currency) {
            case Currency.Polkadot:
                this.logo = require("assets/img/dot.png")
                this.symbol = "DOT"
                break
            case Currency.Kusama:
                this.logo = require("assets/img/kusama.png")
                this.symbol = "KSM"
                break
        }

    }
}
