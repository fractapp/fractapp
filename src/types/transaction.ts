import {Currency} from './/wallet';

/**
 * Transaction type
 * @category Models
 */
export enum TxType {
  None = 0,
  Sent,
  Received,
}

/**
 * Transaction action type
 * @category Models
 */
export enum TxAction {
  Transfer,
  StakingReward,
  StakingCreateWithdrawalRequest,
  StakingWithdrawn,
  StakingOpenDeposit,
  StakingAddAmount,
  ConfirmWithdrawal,
  UpdateNomination
}

export enum TxStatus {
  Pending = 0,
  Success,
  Fail,
}

/**
 * @category Models
 */
export type Transaction = {
  id: string;
  hash: string;
  userId: string | null;

  address: string;
  currency: Currency;
  action: TxAction,
  txType: TxType;
  timestamp: number;

  value: number;
  planckValue: string;
  usdValue: number;
  fullValue: string;

  fee: number;
  planckFee: string;
  usdFee: number;

  status: TxStatus;
};


/**
 * @category Models
 */
export type ApiTransaction = {
  id: string;
  hash: string;
  currency: Currency;

  from: string;
  userFrom: string;

  action: TxAction,

  to: string;
  userTo: string;

  value: string;
  fee: string;
  price: number

  timestamp: number;
  status: TxStatus;
};

