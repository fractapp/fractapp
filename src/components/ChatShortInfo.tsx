import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import {WalletLogo} from 'components/WalletLogo';
import {getSymbol} from 'models/wallet';
import {Transaction, TxType} from 'models/transaction';
import stringUtils from 'utils/string';

/**
 * @category Components
 */
export const ChatShortInfo = ({
  name,
  notificationCount,
  tx,
}: {
  name: string;
  notificationCount: number;
  tx: Transaction;
}) => {
  const now = new Date();
  return (
    <View style={styles.account}>
      <View
        style={{
          flex: 1,
          flexDirection: 'row',
          alignItems: 'center',
          marginLeft: '4%',
          marginRight: '3%',
        }}>
        <WalletLogo currency={tx.currency} size={60} />
        <View style={{flex: 1, flexDirection: 'column', marginLeft: 10}}>
          <View style={{height: 25, flexDirection: 'row'}}>
            <View style={{flex: 3}}>
              <Text style={[styles.name, {textAlign: 'left'}]}>
                {stringUtils.formatNameOrAddress(name)}
              </Text>
            </View>
            <View style={{flex: 1}}>
              <Text style={[styles.time, {textAlign: 'right'}]}>
                {stringUtils.forChatInfo(now, new Date(tx.timestamp))}
              </Text>
            </View>
          </View>
          <View style={{height: 25, flexDirection: 'row'}}>
            <View style={{flex: 8}}>
              <Text style={[styles.msg, {textAlign: 'left'}]}>
                {tx.txType === TxType.Sent
                  ? `Sent -$${tx.usdValue} (${tx.value} ${getSymbol(
                      tx.currency,
                    )})`
                  : `Received +$${tx.usdValue} (${tx.value} ${getSymbol(
                      tx.currency,
                    )})`}
              </Text>
            </View>
            <View style={{flex: 1, alignItems: 'flex-end'}}>
              <View style={styles.notification}>
                {notificationCount !== 0 ? (
                  <Text style={[styles.notificationText]}>
                    {notificationCount}
                  </Text>
                ) : (
                  <View />
                )}
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  account: {
    height: 80,
    width: '100%',
  },
  name: {
    fontSize: 15,
    fontFamily: 'Roboto-Bold',
    fontStyle: 'normal',
    fontWeight: 'normal',
    color: 'black',
  },
  msg: {
    fontSize: 15,
    fontFamily: 'Roboto-Regular',
    fontStyle: 'normal',
    fontWeight: 'normal',
    color: '#888888',
  },
  time: {
    fontSize: 13,
    fontFamily: 'Roboto-Regular',
    fontStyle: 'normal',
    color: '#888888',
  },
  notification: {
    backgroundColor: '#C4C4C4',
    paddingLeft: 8,
    paddingRight: 8,
    borderRadius: 20,
  },
  notificationText: {
    fontSize: 13,
    fontFamily: 'Roboto-Regular',
    fontStyle: 'normal',
    color: 'white',
  },
});
