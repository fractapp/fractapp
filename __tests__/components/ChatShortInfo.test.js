import React from 'react';
import renderer from 'react-test-renderer';
import {ChatShortInfo} from 'components/ChatShortInfo';
import {Currency} from '../../src/types/wallet';
import {TxStatus, TxType} from '../../src/types/transaction';

jest.mock('utils/backend', () => ({
  getImgUrl: jest.fn(),
}));
it('Test one', () => {
  const tree = renderer
    .create(
      <ChatShortInfo
        name={'name'}
        notificationCount={10}
        tx={{
          id: '4',
          userId: 'userId',
          address: '1exaAg2VJRQbyUBAeXcktChCAqjVP9TUxF3zo23R2T6EGdE',
          currency: Currency.Kusama,
          txType: TxType.Received,
          timestamp: new Date('02-12-2020').getTime(),
          value: 10,
          usdValue: 0,
          fee: 10,
          usdFee: 10,
          status: TxStatus.Pending,
        }}
        user={null}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test two', () => {
  const tree = renderer
    .create(
      <ChatShortInfo
        name={'name'}
        notificationCount={0}
        tx={{
          id: '4',
          userId: 'userId',
          address: '1exaAg2VJRQbyUBAeXcktChCAqjVP9TUxF3zo23R2T6EGdE',
          currency: Currency.Polkadot,
          txType: TxType.Sent,
          timestamp: new Date('02-12-2020').getTime(),
          value: 10,
          usdValue: 0,
          fee: 10,
          usdFee: 10,
          status: TxStatus.Pending,
        }}
        user={{
          id: 'id',
          name: 'name',
          username: 'username',
          avatarExt: 'png',
          lastUpdate: new Date('12-12-2020').getTime(),
          addresses: {
            0: 'addressOne',
            1: 'addressTwo',
          },
        }}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test three', () => {
  const tree = renderer
    .create(
      <ChatShortInfo
        name={'name'}
        notificationCount={10}
        tx={{
          id: '4',
          userId: 'userId',
          address: '1exaAg2VJRQbyUBAeXcktChCAqjVP9TUxF3zo23R2T6EGdE',
          currency: Currency.Kusama,
          txType: TxType.Received,
          timestamp: new Date('02-12-2020').getTime(),
          value: 10,
          usdValue: 0,
          fee: 10,
          usdFee: 10,
          status: TxStatus.Pending,
        }}
        user={{
          id: 'id',
          name: 'name',
          username: 'username',
          avatarExt: '',
          lastUpdate: new Date('12-12-2020').getTime(),
          addresses: {
            0: 'addressOne',
            1: 'addressTwo',
          },
        }}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
