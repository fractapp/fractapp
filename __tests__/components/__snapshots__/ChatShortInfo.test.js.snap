// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test one 1`] = `
<View
  style={
    Object {
      "height": 75,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "marginLeft": "4%",
        "marginRight": "3%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/img/dot.png",
        }
      }
      style={
        Object {
          "borderColor": "black",
          "borderRadius": 27.5,
          "borderWidth": 0.8,
          "height": 55,
          "overflow": "hidden",
          "width": 55,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "marginLeft": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 25,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 10,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "black",
                  "fontFamily": "Roboto-Bold",
                  "fontSize": 15,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {
                  "textAlign": "left",
                },
              ]
            }
          >
            address#1
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 2,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#888888",
                  "fontFamily": "Roboto-Regular",
                  "fontSize": 12,
                  "fontStyle": "normal",
                },
                Object {
                  "textAlign": "center",
                },
              ]
            }
          >
            NaN/NaN/Nundefined
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 23,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 10,
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#888888",
                  "fontFamily": "Roboto-Regular",
                  "fontSize": 14,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {
                  "textAlign": "left",
                },
              ]
            }
          >
            Test
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 2,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#C4C4C4",
                "borderRadius": 20,
                "paddingLeft": 8,
                "paddingRight": 8,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 12,
                    "fontStyle": "normal",
                  },
                ]
              }
            >
              10
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Test three 1`] = `
<View
  style={
    Object {
      "height": 75,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "marginLeft": "4%",
        "marginRight": "3%",
      }
    }
  >
    <Image
      height={55}
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "borderRadius": 30,
          "height": 55,
          "width": 55,
        }
      }
      width={55}
    />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "marginLeft": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 25,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 10,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "black",
                  "fontFamily": "Roboto-Bold",
                  "fontSize": 15,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {
                  "textAlign": "left",
                },
              ]
            }
          >
            user
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 2,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#888888",
                  "fontFamily": "Roboto-Regular",
                  "fontSize": 12,
                  "fontStyle": "normal",
                },
                Object {
                  "textAlign": "center",
                },
              ]
            }
          >
            NaN/NaN/Nundefined
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 23,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 10,
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#888888",
                  "fontFamily": "Roboto-Regular",
                  "fontSize": 14,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {
                  "textAlign": "left",
                },
              ]
            }
          >
            Test3
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 2,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#C4C4C4",
                "borderRadius": 20,
                "paddingLeft": 8,
                "paddingRight": 8,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 12,
                    "fontStyle": "normal",
                  },
                ]
              }
            >
              10
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Test two 1`] = `
<View
  style={
    Object {
      "height": 75,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "marginLeft": "4%",
        "marginRight": "3%",
      }
    }
  >
    <Image
      height={55}
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "borderRadius": 30,
          "height": 55,
          "width": 55,
        }
      }
      width={55}
    />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "marginLeft": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 25,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 10,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "black",
                  "fontFamily": "Roboto-Bold",
                  "fontSize": 15,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {
                  "textAlign": "left",
                },
              ]
            }
          >
            user
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 2,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#888888",
                  "fontFamily": "Roboto-Regular",
                  "fontSize": 12,
                  "fontStyle": "normal",
                },
                Object {
                  "textAlign": "center",
                },
              ]
            }
          >
            NaN/NaN/Nundefined
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 23,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 10,
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#888888",
                  "fontFamily": "Roboto-Regular",
                  "fontSize": 14,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {
                  "textAlign": "left",
                },
              ]
            }
          >
            Test2
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 2,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#C4C4C4",
                "borderRadius": 20,
                "paddingLeft": 8,
                "paddingRight": 8,
              }
            }
          >
            <View />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
